misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/ReferenceType.java

```diff
@@ -164,20 +184,38 @@ public static RefType fromSemanticSymbol(TypeSymbol symbol, String name, String
                     ? typeRefSymbol.getModule().get().id().toString()
                     : null;
             return fromSemanticSymbol(typeSymbol, name, moduleId);
-        } else if (kind == TypeDescKind.INT) {
-            return new RefType("int");
+        }  else if (kind == TypeDescKind.INT) {
+            RefType refType = new RefType("int");
+            refType.typeName = "int";
+            return refType;
         } else if (kind == TypeDescKind.STRING) {
-            return new RefType("string");
+            RefType refType = new RefType("string");
+            refType.typeName = "string";
+            return refType;
         } else if (kind == TypeDescKind.FLOAT) {
-            return new RefType("float");
+            RefType refType = new RefType("float");
+            refType.typeName = "float";
+            return refType;
         } else if (kind == TypeDescKind.BOOLEAN) {
-            return new RefType("boolean");
+            RefType refType = new RefType("boolean");
+            refType.typeName = "boolean";
+            return refType;
         } else if (kind == TypeDescKind.NIL) {
-            return new RefType("nil");
+            RefType refType = new RefType("nil");
+            refType.typeName = "nil";
+            return refType;
         } else if (kind == TypeDescKind.DECIMAL) {
-            return new RefType("decimal");
+            RefType refType = new RefType("decimal");
+            refType.typeName = "decimal";
+            return refType;
```

Comment:
How about using another `constructor` to initialize the refType with `new RefType("decimal", "decimal")`. Then, you don't have to set `typeName` separately .